software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -156,6 +157,34 @@ public void install() {
                 .add("ln -s `which riak-admin` " + Urls.mergePaths(installBin, "riak-admin"))
                 .build();
     }
+    
+    private List<String> installPackageCloud() {
+        return ifExecutable0Else1("yum", installDebianBased(), installRpmBased());
+    }
+
+    private ImmutableList<String> installDebianBased() {
+        ImmutableList.Builder<String> commands = ImmutableList.<String>builder();
+        commands.add("curl https://packagecloud.io/install/repositories/basho/riak/script.deb | sudo bash");
+        commands.add("sudo apt-get install --assume-yes riak");
+        return commands.build();
+    }
+    
+    private ImmutableList<String> installRpmBased() {
+        ImmutableList.Builder<String> commands = ImmutableList.<String>builder();
+        commands.add("curl https://packagecloud.io/install/repositories/basho/riak/script.rpm | sudo bash");
+        commands.add("sudo yum install -y riak");
```

Comment:
Suggest you use `.add(BashCommands.sudo("yum install -y riak"))`. If running as the root user, then on some VMs it causes problems to try to execute `sudo`. The `sudo(...)` method gets around that by checking if the user is already root.

Same applies for other uses of sudo.